src/Binary/binarySerializer.cpp

```diff
@@ -139,10 +139,138 @@ void binary::BinarySerializer::serializeMethod(::Meta::MethodMeta* meta, binary:
+                    (this->cachedSignatures[i].types[x] == FFIDouble && signature[x]->getType() == ::Meta::TypeDouble) ||
+                    (this->cachedSignatures[i].types[x] == FFIStruct && signature[x]->getType() == ::Meta::TypeStruct) ||
+                    (this->cachedSignatures[i].types[x] == FFIPointer && (signature[x]->getType() == ::Meta::TypePointer ||
+                                                                          signature[x]->getType() == ::Meta::TypeEnum ||
+                                                                          signature[x]->getType() == ::Meta::TypeClass ||
+                                                                          signature[x]->getType() == ::Meta::TypeId ||
+                                                                          signature[x]->getType() == ::Meta::TypeBlock ||
+                                                                          signature[x]->getType() == ::Meta::TypeUnion ||
+                                                                          signature[x]->getType() == ::Meta::TypeStruct ||
+                                                                          signature[x]->getType() == ::Meta::TypeSelector ||
+                                                                          signature[x]->getType() == ::Meta::TypeProtocol ||
+                                                                          signature[x]->getType() == ::Meta::TypeInterface ||
+                                                                          signature[x]->getType() == ::Meta::TypeInstancetype ||
+                                                                          signature[x]->getType() == ::Meta::TypeVaList ||
+                                                                          signature[x]->getType() == ::Meta::TypeTypeArgument ||
+                                                                          signature[x]->getType() == ::Meta::TypeConstantArray ||
+                                                                          signature[x]->getType() == ::Meta::TypeFunctionPointer ||
+                                                                          signature[x]->getType() == ::Meta::TypeBridgedInterface ||
+                                                                          signature[x]->getType() == ::Meta::TypeCString))) {
+                    if (x == cachedSignatures[i].types.size()-1) {
+                        return cachedSignatures[i].offset;
+                    }
+                }  else {
+                    break;
+                }
+                
+            }
+        }
+    }
+    
+    binary::MetaFileOffset newOffset = this->typeEncodingSerializer.visit(signature);
+    CachedSignature newSignature;
+    newSignature.offset = newOffset;
+    
+    for (size_t i = 0; i < signature.size(); i++) {
```

Comment:
Since we are mapping `Meta::Type` objects to `FFIType` values for a second time in the same method I suggest extracting the mapping logic outside of the binary serialization. The mapping logic is not directly related to the binary serialization, it is a property of the `Meta::Type` itself. I suggest introducing an instance method of the `Meta::Type` structure, mapping the type to a `FFIType`:

```
FFIType toFFIType() const
{
    switch (this->type) {
        case TypeBool:
            return FFYTypeUInt8
        case TypeShort:
            ...
}
```

Then a `MethodMeta` can have a method that converts a vector of `Meta::Type`s to vector of `FFIType`s e.g.:

```
std::vector<FFIType> getFFISignature() const
{
    ...
}
```

This way the mapping logic will not be extracted from the binary serializer and can be used in other modules without code repetition e.g. it is a good idea to print the FFI signature of a method in the Yaml output.